internal/leadership/leadership.go

```diff
@@ -66,22 +72,91 @@ func NewLeaderElection() *LeaderElection {
 
 // createLeadershipTable creates the database table used for leader election
 func (l *LeaderElection) createLeadershipTable() error {
-	var statement string
-	switch l.storageProvider {
-	case string(storage.POSTGRESQL):
-		statement = "CREATE TABLE IF NOT EXISTS members (id TEXT PRIMARY KEY, registration NUMERIC, heartbeat NUMERIC)"
-	case string(storage.MYSQL):
-		statement = "CREATE TABLE IF NOT EXISTS members (id VARCHAR(50) PRIMARY KEY, registration BIGINT, heartbeat BIGINT)"
-	case string(storage.SQLITE):
-		statement = "CREATE TABLE IF NOT EXISTS members (id TEXT PRIMARY KEY, registration INTEGER, heartbeat INTEGER)"
+	switch l.storageType {
+
+	// SQL Adapter
+	case string(storage.SQL):
+		var statement string
+		switch l.storageProvider {
+		case string(storage.POSTGRESQL):
+			statement = "CREATE TABLE IF NOT EXISTS members (id TEXT PRIMARY KEY, registration NUMERIC, heartbeat NUMERIC)"
```

Comment:
Consider extracting the table and field names as constants and use `fmt` to format the queries.
This will reduce the change of error when editing them

Reply:
I'll do this one in a separate commit as it touches a bunch of things and not strictly related to this feature.